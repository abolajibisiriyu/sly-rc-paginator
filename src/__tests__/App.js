import React from "react";
import Paginator from "../index";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

let paginatorMeta, paginatorOptions, onPageChanged;

beforeAll(() => {
  paginatorMeta = {
    totalItems: 20,
    currentPage: 1,
    itemsPerPage: 10
  };
  paginatorOptions = {
    ulClassName: "ulClassName",
    liClassName: "liClassName",
    activeClassName: "activeClassName",
    disabledClassName: "disabledClassName",
    anchorClassName: "anchorClassName"
  };
  onPageChanged = jest.fn(nextPage => nextPage);
});

describe("Paginator Component", () => {
  test("renders correctly", () => {
    const paginatorTree = renderer
      .create(
        <Paginator
          meta={paginatorMeta}
          options={paginatorOptions}
          onPageChange={onPageChanged}
        />
      )
      .toJSON();
    expect(paginatorTree).toMatchSnapshot();
  });

  test("renders a list of links", () => {
    const wrapper = shallow(
      <Paginator
        meta={paginatorMeta}
        options={paginatorOptions}
        onPageChange={onPageChanged}
      />
    );
    expect(wrapper.find("ul")).toHaveLength(1);
    expect(wrapper.find("li").length).toBeGreaterThan(4);
    wrapper.find("li").forEach(node => {
      expect(node.find("a")).toHaveLength(1);
    });
  });

  test("renders a list li with className of paginatorOptions liClassName", () => {
    const wrapper = shallow(
      <Paginator
        meta={paginatorMeta}
        options={paginatorOptions}
        onPageChange={onPageChanged}
      />
    );
    wrapper.find("li").forEach(node => {
      expect(node.prop("className")).toContain(paginatorOptions.liClassName);
    });
  });
});
