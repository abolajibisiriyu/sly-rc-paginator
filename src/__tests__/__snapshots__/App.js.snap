// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginator Component renders correctly 1`] = `
<ul
  className="ulClassName"
>
  <li
    className="liClassName disabledClassName"
  >
    <a
      className="anchorClassName"
      onClick={[Function]}
    >
      First
    </a>
  </li>
  <li
    className="liClassName disabledClassName"
  >
    <a
      className="anchorClassName"
      onClick={[Function]}
    >
      Previous
    </a>
  </li>
  <li
    className="liClassName activeClassName"
  >
    <a
      className="anchorClassName"
      onClick={[Function]}
    >
      1
    </a>
  </li>
  <li
    className="liClassName "
  >
    <a
      className="anchorClassName"
      onClick={[Function]}
    >
      2
    </a>
  </li>
  <li
    className="liClassName "
  >
    <a
      className="anchorClassName"
      onClick={[Function]}
    >
      Next
    </a>
  </li>
  <li
    className="liClassName "
  >
    <a
      className="anchorClassName"
      onClick={[Function]}
    >
      Last
    </a>
  </li>
</ul>
`;
